pkg/kubelet/envvars/envvars.go

```diff
@@ -34,16 +33,14 @@ func FromServices(services []*v1.Service) []v1.EnvVar {
 	for i := range services {
 		service := services[i]
 
-		// ignore services where ClusterIP is "None" or empty
-		// the services passed to this method should be pre-filtered
-		// only services that have the cluster IP set should be included here
-		if !v1helper.IsServiceIPSet(service) {
-			continue
-		}
-
 		// Host
 		name := makeEnvVariableName(service.Name) + "_SERVICE_HOST"
-		result = append(result, v1.EnvVar{Name: name, Value: service.Spec.ClusterIP})
+		if service.Spec.Type == v1.ServiceTypeExternalName {
+			result = append(result, v1.EnvVar{Name: name, Value: service.Spec.ExternalName})
```

Comment:
This seems like it would have been a huge bug with external name services, I guess no one used them and the env vars at the same time.

The variable is called "_HOST" not "_IP", so I think this is probably OK.

The alternative would be to have kubelet do a DNS resolution here, but that defeats the purpose of having the name be dynamic.

If this breaks something for people, they can avoid it by not using an ExternalName type service. I think it deserves a release note but is OK.